pkgs/applications/editors/vim/plugins/vim-utils.nix

```diff
@@ -154,13 +154,18 @@ let
     # make sure all the dependencies of the plugin are also derivations
     drv // { dependencies = map (pluginToDrv knownPlugins) (drv.dependencies or [ ]); };
 
-  # transitive closure of plugin dependencies (plugin needs to be a derivation)
-  transitiveClosure =
-    plugin:
-    [ plugin ]
-    ++ (lib.unique (builtins.concatLists (map transitiveClosure plugin.dependencies or [ ])));
-
-  findDependenciesRecursively = plugins: lib.concatMap transitiveClosure plugins;
+  findDependenciesRecursively =
+    plugins:
+    if plugins == [ ] then
+      [ ]
+    else
+      let
+        # reverse so lib.unique prefers the last duplicate plugin instead of the first one
```

Comment:
why is it better to select the last rather than the first ?

Reply:
It seems linkFarm only keeps the last entry with the same path.

Reply:
but your fix might break other people setup where it's more important to get the latter. So isn't the proper fix to warn in case 2 files conflict ?

Reply:
I fixed the problem that the directly specified plugin is not installed but the dependency is kept.

Reply:
Because if we would prefer first element first, we overwrite our plugins with dependencies of other plugins. I tried to come up with an example, but evaluation of this function is too weird for me to understand.

I cannot come up for why you would want your custom plugins to be ignored, while we have [a good use case](https://github.com/NixOS/nixpkgs/pull/319233#issue-2347899631) for why you don't